framework/docker/da_node.go

```diff
@@ -330,6 +341,58 @@ func (n *DANode) getImage() DockerImage {
 	return n.Image
 }
 
+// getRPCPort returns the RPC port for this node, with per-node config taking priority over network defaults
+func (n *DANode) getRPCPort() string {
+	if nodeConfig := n.getNodeConfig(); nodeConfig != nil && nodeConfig.RPCPort != "" {
+		return nodeConfig.RPCPort + "/tcp"
+	}
+	if n.cfg.DataAvailabilityNetworkConfig.DefaultRPCPort != "" {
+		return n.cfg.DataAvailabilityNetworkConfig.DefaultRPCPort + "/tcp"
+	}
+	return "26658/tcp"
+}
+
+// getP2PPort returns the P2P port for this node, with per-node config taking priority over network defaults
+func (n *DANode) getP2PPort() string {
+	if nodeConfig := n.getNodeConfig(); nodeConfig != nil && nodeConfig.P2PPort != "" {
+		return nodeConfig.P2PPort + "/tcp"
+	}
+	if n.cfg.DataAvailabilityNetworkConfig.DefaultP2PPort != "" {
+		return n.cfg.DataAvailabilityNetworkConfig.DefaultP2PPort + "/tcp"
+	}
+	return "2121/tcp"
+}
+
+// getCoreRPCPort returns the core RPC port this node should connect to
+func (n *DANode) getCoreRPCPort() string {
+	if nodeConfig := n.getNodeConfig(); nodeConfig != nil && nodeConfig.CoreRPCPort != "" {
+		return nodeConfig.CoreRPCPort
+	}
+	if n.cfg.DataAvailabilityNetworkConfig.DefaultCoreRPCPort != "" {
+		return n.cfg.DataAvailabilityNetworkConfig.DefaultCoreRPCPort
+	}
+	return "26657"
```

Comment:
maybe for these return values we can use the constants (change them from being nat.Port to just port)